server/py/framework/db/sqldb/models.py

```diff
@@ -1047,27 +1047,132 @@ def get_identifier_string(self) -> str:
             return f"{self.key}"
 
 
-@event.listens_for(AlertActivation.__table__, "before_create")
-def _disable_autoinc_on_sqlite(table, connection, **kw):
-    if connection.dialect.name.startswith(mlrun.common.db.dialects.Dialects.SQLITE):
-        # disable SQLAlchemy's AUTOINCREMENT flag
-        table.c.id.autoincrement = False
+def event_listen_for_dialects(
+    target: Any,
+    identifier: str,
+    relevant_dialects: list[str],
+) -> Callable:
+    """
+    Register an SQLAlchemy event listener that runs only for the chosen dialects.
+    Works for `before_insert` (mapper events) and `after_create` (DDL events).
+    """
+    relevant = set(relevant_dialects)
```

Comment:
nitpick but use full names
```suggestion
    relevant_dialects = set(relevant_dialects)
```